Dockerfile

```diff
@@ -0,0 +1,71 @@
+FROM quay.io/keboola/aws-cli
+ARG AWS_SECRET_ACCESS_KEY
+ARG AWS_ACCESS_KEY_ID
+RUN /usr/bin/aws s3 cp s3://keboola-drivers/teradata/tdodbc1710-17.10.00.08-1.x86_64.deb /tmp/teradata/tdodbc.deb
+RUN /usr/bin/aws s3 cp s3://keboola-drivers/teradata/utils/TeradataToolsAndUtilitiesBase__ubuntu_x8664.17.00.34.00.tar.gz  /tmp/teradata/tdutils.tar.gz
+
+FROM php:7.4-cli
+
+ARG COMPOSER_FLAGS="--prefer-dist --no-interaction"
+ARG DEBIAN_FRONTEND=noninteractive
+ENV COMPOSER_ALLOW_SUPERUSER 1
+ENV COMPOSER_PROCESS_TIMEOUT 7200
+
+WORKDIR /code/
+
+COPY etc/docker/php-prod.ini /usr/local/etc/php/php.ini
+COPY etc/docker/composer-install.sh /tmp/composer-install.sh
+
+RUN apt-get update -q \
+    && ACCEPT_EULA=Y apt-get install -y --no-install-recommends\
+        locales \
+        unzip \
+        ca-certificates \
+        unixodbc \
+        unixodbc-dev \
+	&& rm -r /var/lib/apt/lists/* \
+	&& sed -i 's/^# *\(en_US.UTF-8\)/\1/' /etc/locale.gen \
+	&& locale-gen \
+	&& chmod +x /tmp/composer-install.sh \
+	&& /tmp/composer-install.sh
+
+ENV LANGUAGE=en_US.UTF-8
+ENV LANG=en_US.UTF-8
+ENV LC_ALL=en_US.UTF-8
+
+# Teradata ODBC
+COPY --from=0 /tmp/teradata/tdodbc.deb /tmp/teradata/tdodbc.deb
+COPY etc/docker/teradata/odbc.ini /tmp/teradata/odbc_td.ini
+COPY etc/docker/teradata/odbcinst.ini /tmp/teradata/odbcinst_td.ini
+
+RUN dpkg -i /tmp/teradata/tdodbc.deb \
+    && cat /tmp/teradata/odbc_td.ini >> /etc/odbc.ini \
+    && cat /tmp/teradata/odbcinst_td.ini >> /etc/odbcinst.ini \
+    && rm -r /tmp/teradata \
+    && docker-php-ext-configure pdo_odbc --with-pdo-odbc=unixODBC,/usr \
+    && docker-php-ext-install pdo_odbc
+
+ENV ODBCHOME = /opt/teradata/client/ODBC_64/
+ENV ODBCINI = /opt/teradata/client/ODBC_64/odbc.ini
+ENV ODBCINST = /opt/teradata/client/ODBC_64/odbcinst.ini
+ENV LD_LIBRARY_PATH = /opt/teradata/client/ODBC_64/lib
+
+# Teradata Utils
+COPY --from=0 /tmp/teradata/tdutils.tar.gz /tmp/teradata/tdutils.tar.gz
+RUN cd /tmp/teradata \
+    && tar -xvaf tdutils.tar.gz \
+    && sh /tmp/teradata/TeradataToolsAndUtilitiesBase/.setup.sh tptbase s3axsmod \
+    && rm -rf /var/lib/apt/lists/* \
+    && rm -rf /tmp/teradata
+
+## Composer - deps always cached unless changed
+# First copy only composer files
+COPY composer.* /code/
+# Download dependencies, but don't run scripts or init autoloaders as the app is missing
+RUN composer install $COMPOSER_FLAGS --no-scripts --no-autoloader
```

Comment:
Nejde to zbuildit:
```------                                                                                                                                                                                                                                       > [stage-1 13/15] RUN composer install --prefer-dist --no-interaction --no-scripts --no-autoloader:
#21 0.445 No composer.lock file present. Updating dependencies to latest instead of installing from lock file. See https://getcomposer.org/install for more information.                                                                    #21 0.445 Loading composer repositories with package information
#21 1.684 Updating dependencies                                                                                                                                                                                                             #21 1.694 Your requirements could not be resolved to an installable set of packages.
#21 1.694
#21 1.694   Problem 1
#21 1.694     - keboola/table-backend-utils v0.18.0 requires ext-odbc * -> it is missing from your system. Install or enable PHP's odbc extension.
#21 1.694     - keboola/db-import-export dev-master requires keboola/table-backend-utils ^0.18 -> satisfiable by keboola/table-backend-utils[v0.18.0].
#21 1.694     - Root composer.json requires keboola/db-import-export dev-master -> satisfiable by keboola/db-import-export[dev-master].
#21 1.694
#21 1.694 To enable extensions, verify that they are enabled in your .ini files:
#21 1.694     - /usr/local/etc/php/php.ini
#21 1.694     - /usr/local/etc/php/conf.d/docker-php-ext-pdo_odbc.ini
#21 1.694     - /usr/local/etc/php/conf.d/docker-php-ext-sodium.ini
#21 1.694 You can also run `php --ini` in a terminal to see which files are used by PHP in CLI mode.
#21 1.694 Alternatively, you can run Composer with `--ignore-platform-req=ext-odbc` to temporarily ignore these required extensions.```

Reply:
jo on je funkční až v protobuf PR :/ cherypicknu to.

Reply:
jo, uz to proslo